packages/components/src/Notification/ToastNotification/context/ToastNotificationContext.tsx

```diff
@@ -44,18 +44,23 @@ export const ToastNotificationProvider = ({
   const [notifications, setNotifications] = useState<ToastNotificationObj[]>([])
 
   const addToastNotification: ToastNotificationContextValue["addToastNotification"] =
-    notification => {
-      const uuid = uuidv4()
-      const notificationWithId = { id: uuid, ...notification }
-
-      const notificationExists = notifications.find(
-        ({ id }) => id === notification.id
-      )
-
-      if (!notificationExists) {
-        setNotifications(existing => [...existing, notificationWithId])
-      }
-    }
+    useCallback(
```

Comment:
Wrapping useCallback around this because in the consumer it could be used as a dependancy of a useEffect